torch/_inductor/kernel/flex_attention.py

```diff
@@ -374,6 +375,46 @@ def load_checked_2d(
     off_zq = tl.program_id(1) // HQ
     off_hq = tl.program_id(1) % HQ
 
+
+    # Setting up the TMA descriptors for Q, K, V
+    desc_q = None
+    desc_k = None
+    desc_v = None
+    if USE_TMA:
+        TMA_SIZE = 128
+        workspace_base = ws_ptr + q_start * 3 * TMA_SIZE
```

Comment:
we need to ensure the descriptor base is unique. We have 
B * H * Q/BLOCK_M programs
and then 3 per descriptors per program
So maybe we think about this as a 

B*H x  Q/BLOCK_M  x 3*TMA_size Array

So something like 
desc_m_stride = 3 * TMA_SIZE
BH_stride = tl.num_programs(0) * desc_m_stride

BH_NUM = program_id(1)
workspace_base = ws_ptr + BH_NUM * BH_STRIDE + q_start *desc_m_stride 
desc_q ...

S

Reply:
In essence each program will have a unique set descriptors so there isn't any race condition to access the same TMA memory across batches.